pyomo/contrib/solver/solvers/sol_reader.py

```diff
@@ -133,12 +139,19 @@ def get_duals(
                 "have happened. Report this error to the Pyomo Developers."
             )
         res = {}
-        if self._nl_info.scaling is None:
-            scale_list = [1] * len(self._nl_info.constraints)
-            obj_scale = 1
+        # Set a default scaling
```

Comment:
I have concerns about this implementation.  99% of the time, the user us going to load everything, and 95% of the non-IDAES users will never use scaling.  Unfortunately, this implementation incurs the overhead of the filter and the division regardless of what the user is doing.  I would suggest a different approach:
```python
scaling = self._nl_info.scaling
if scaling:
    _iter = zip(self._nl_info.constraints, self._sol_data.duals, scaling.constraints)
    obj_scale = self._nl_info.scaling.objectives[0]
else:
    _iter = zip(self._nl_info.constraints, self._sol_data.duals)
if cons_to_load is not None:
    _iter = filter(lambda x: x[0] in cons_to_load, _iter)
if scaling:
    res = {con: val * scale / obj_scale for con, val, scale in _iter}
else:
    res = {con: val for con, val in _iter}